pandas/core/indexes/base.py

```diff
@@ -2425,11 +2450,22 @@ def difference(self, other, sort=True):
         Parameters
         ----------
         other : Index or array-like
-        sort : bool, default True
-            Sort the resulting index if possible
+        sort : False or None, default None
+            Whether to sort the resulting index. By default, the
+            values are attempted to be sorted, but any TypeError from
+            incomparable elements is caught by pandas.
+
+            * None : Attempt to sort the result, but catch any TypeErrors
+              from comparing incomparable elements.
+            * False : Do not sort the result.
 
             .. versionadded:: 0.24.0
 
+            .. versionchanged:: 0.24.1
+
+               Changed `True` to `None`, which matches the behavior of
```

Comment:
May not be obvious to users what "Changed True to None" means. Maybe something like "Changed the default from True to False, and disallowed True?" That's not terribly clear either.

Reply:
Changed it to:

```
            .. versionchanged:: 0.24.1

               Changed the default value from ``True`` to ``None``
               (without change in behaviour).
```